src/dxf-handler.ts

```diff
@@ -216,70 +212,36 @@ export class DxfHandler {
         return lines;
     }
 
-    _scaleData(): [number[], number] {
-        const range = this.currentLayer.maxPoint[0] - this.currentLayer.minPoint[0];
+    _scaleData(layer: Layer): [number[], number] {
+        const range = layer.maxPoint[0] - layer.minPoint[0];
 
         // Avoid division by zero
         if (range === 0) {
-            return [this.currentLayer.lines, 1]; // No scaling needed
+            return [layer.lines, 1]; // No scaling needed
         }
 
         const scaleFactor = 1 / range;
 
         console.log(scaleFactor);
         const res: number[] = [];
-        for (let i = 0; i < this.currentLayer.lines.length; i++) {
-            res.push(this.currentLayer.lines[i] * scaleFactor);
+        for (let i = 0; i < layer.lines.length; i++) {
+            res.push(layer.lines[i] * scaleFactor);
         }
         return [res, scaleFactor];
     }
 
-    /**
-     * Retrieves the current layer.
-     *
-     *  @note
-     * The returned array is flattened for cache coherency. Iterate over it in steps of 4 to access individual line coordinates.
-     *
-     *  @example
-     * ```js
-     * const list = handler.currentLayer.lines;
-     * let i = 0;
-     * while (i < list.length) {
-     *   const line_start_x = list[i];
-     *   const line_start_y = list[i + 1];
-     *   const line_end_x = list[i + 2];
-     *   const line_end_y = list[i + 3];
-     *   i += 4;
-     * }
-     * ```
-     * @see Layer
-     *
-     *  @returns An array of numbers, aligned as follows: [start_x0, start_y0, end_x1, end_y1, ...]
-     */
-    get currentLayer(): Layer {
-        if (this.layers.length === 0)
-            return {
-                name: '',
-                lines: [],
-                minPoint: new Float32Array([0, 0]),
-                maxPoint: new Float32Array([0, 0]),
-            };
-        return this.layers[this.#currentIndex];
-    }
-
-    moveToNextLayer() {
-        this.#currentIndex = (this.#currentIndex + 1) % this.layers.length;
-    }
-
-    moveToPrevLayer() {
-        this.#currentIndex =
-            this.#currentIndex === 0 ? this.layers.length - 1 : this.#currentIndex - 1;
-    }
-
-    /**
-     * Gets the index of the current layer.
-     */
-    get currentIndex(): number {
-        return this.#currentIndex;
+    mergeLayers(layerIndices: number[]) {
+        const mergedLayer: Layer = {
```

Comment:
instead of creating a new Layer, just return a array of the layers of all the indices that are provided and make the receiving function iterate through the received references.

Reply:
Have to check this